### PR TITLE
Add a better check for ztunnel pod guess

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**

### CODE REVIEW

File: istioctl/cmd/proxyconfig.go [hunk 1]
```diff
@@ -164,6 +164,14 @@ var (
 	reset             = false
 )
 
+var isZtunnelPod = func(podName, podNamespace string) bool {
+	client, err := kubeClient(kubeconfig, configContext)
+	if err != nil {
+		return ambientutil.IsZtunnelPod(nil, podName, podNamespace)
+	}
+	return ambientutil.IsZtunnelPod(client, podName, podNamespace)
```
Comment (bleggett) [inline istioctl/cmd/proxyconfig.go:172]:
```suggestion
	client, err := kubeClient(kubeconfig, configContext)
	return ambientutil.IsZtunnelPod(client, podName, podNamespace)
```

I can definitely understand the argument that relying the label is better than relying on the podname prefix.

But why not do this? Under what valid scenarios would 

1. We expect kubeClient creation to fail?
2. We want to silently fall back on a weaker detection method if it does?

Comment (hanxiaop) [reply istioctl/cmd/proxyconfig.go:172]: > 1\. We expect kubeClient creation to fail?

KubeClient creation might fail in cases such as invalid config/contexts or connectivity issues. However, it is not very common when using istioctl.

> 2\. We want to silently fall back on a weaker detection method if it does?

I think your suggestion is more predictable and concise compared to silently falling back on a weaker detection method. Thanks.

**Review by bleggett — APPROVED** [review]
LGTM

**Comment by hanxiaop** [comment]
@istio/wg-test-and-release-maintainers PTAL, thanks

**Review by ericvn — APPROVED** [review]
LGTM

**Pull request merged**